pyrobosim/pyrobosim/core/locations.py

```diff
@@ -17,22 +18,26 @@ class Location:
     """Representation of a location in the world."""
 
     # Default class attributes
-    metadata = EntityMetadata()
+    metadata = EntityMetadata(metadata_type="locations")
     """ Metadata for location categories. """
     height = 1.0
     """ Vertical height of location. """
     viz_color = (0, 0, 0)
     """ Visualization color (RGB tuple). """
 
     @classmethod
-    def set_metadata(cls, filename):
+    def add_metadata(cls, filename):
         """
-        Assign a metadata file to the :class:`pyrobosim.core.locations.Location` class.
+        Add metadata from a file to the :class:`pyrobosim.core.locations.Location` class.
 
         :param filename: Path to location metadata YAML file.
         :type filename: str
         """
-        cls.metadata = EntityMetadata(filename)
+        if not os.path.isfile(filename):
+            raise FileNotFoundError(f"Metadata file not found: {filename}")
```

Comment:
Is it better to put this check in the constructor of `EntityMetadata` so more call sites can benefit from this?